Wiener_1923/Convergence.lean

```diff
@@ -0,0 +1,198 @@
+import Mathlib.Probability.Independence.Basic
+import Mathlib.Probability.Distributions.Gaussian
+import Mathlib.MeasureTheory.Constructions.UnitInterval
+import Mathlib.MeasureTheory.Measure.ProbabilityMeasure
+import Mathlib.MeasureTheory.Measure.Portmanteau
+import Batteries.Data.Rat.Basic
+import Mathlib.Probability.IdentDistrib
+import Mathlib.Probability.Independence.Integrable
+import Mathlib.MeasureTheory.Integral.DominatedConvergence
+import Mathlib.Analysis.SpecificLimits.FloorPow
+import Mathlib.Analysis.PSeries
+import Mathlib.Analysis.Asymptotics.SpecificAsymptotics
+import Mathlib.Probability.Variance
+import Mathlib.Topology.Basic
+import Mathlib.Topology.Filter
+import Mathlib.Order.Interval.Set.Monotone
+import Mathlib.Topology.Instances.Irrational
+import Mathlib.Topology.Algebra.Order.Archimedean
+import Mathlib.Topology.Compactness.Paracompact
+import Mathlib.Topology.Metrizable.Urysohn
+import Mathlib.Topology.EMetricSpace.Paracompact
+import Mathlib.Topology.Separation.NotNormal
+import Mathlib.Topology.Baire.Lemmas
+import Mathlib.Topology.Baire.LocallyCompactRegular
+import Mathlib.Data.Complex.Basic
+import Mathlib.Data.Complex.Exponential
+import Mathlib.Analysis.BoxIntegral.Basic
+import Mathlib.MeasureTheory.Integral.BoundedContinuousFunction
+import Mathlib.MeasureTheory.Function.ConditionalExpectation.AEMeasurable
+
+open MeasureTheory ProbabilityTheory NNReal Real Filter Finset Asymptotics
+open Set Filter TopologicalSpace
+open Topology Filter Cardinal MeasureSpace  MeasureTheory.Measure BoundedContinuousFunction BoxIntegral
+
+
+def ConvergenceInDistribution1 {Ω : Type*} [MeasureSpace Ω]
+                               {μ : ProbabilityMeasure Ω}
+                               {X_i : ℕ → Ω → ℝ}
+                               {X_lim : Ω → ℝ}: Prop :=
+  ∀ x : ℝ, Tendsto (fun i => rnDeriv (Measure.map (X_i i) μ) volume x) atTop (𝓝 (rnDeriv (Measure.map X_lim μ) volume x))
+
+def AlmostSureConvergence {Ω : Type*} [MeasureSpace Ω]
```

Comment:
I would have thought a definition for this already exists?

Reply:
I agree, but I haven't found an efficient way to search MathLib. This is also something we can discuss, share some workflows / hacks

Reply:
I couldn't find it either although I found many `Tendsto (fun n => X_i n ω) atTop (𝓝  ((X_lim ω))`. Perhaps we should create a list of things we had to invent and which we couldn't find.